packages/cli/src/utils/build.js

```diff
@@ -290,129 +500,83 @@ const _buildFunc = async ({
+      'zapier-' + crypto.randomBytes(4).toString('hex'),
+    );
+    debug('Building in temp directory: ', workingDir);
   }
 
-  const copyFilter = skipNpmInstall
-    ? (src) => !src.endsWith('.zip')
-    : undefined;
-
-  await copyDir(wdir, tmpDir, { filter: copyFilter });
+  await maybeRunBuildScript({ printProgress });
 
-  if (skipNpmInstall) {
-    const corePackageDir = findCorePackageDir();
-    const nodeModulesDir = path.dirname(corePackageDir);
-    const workspaceRoot = path.dirname(nodeModulesDir);
-    if (wdir !== workspaceRoot) {
-      // If we're in here, it means the user is using npm/yarn workspaces
-      const workspaces = listWorkspaces(workspaceRoot);
-
-      await copyDir(nodeModulesDir, path.join(tmpDir, 'node_modules'), {
-        filter: (src) => {
-          if (src.endsWith('.zip')) {
-            return false;
-          }
-          const stat = fse.lstatSync(src);
-          if (stat.isSymbolicLink()) {
-            const realPath = path.resolve(
-              path.dirname(src),
-              fse.readlinkSync(src),
-            );
-            for (const workspace of workspaces) {
-              // Use minimatch to do glob pattern match. If match, it means the
-              // symlink points to a workspace package, so we don't copy it.
-              if (minimatch(realPath, workspace)) {
-                return false;
-              }
-            }
-          }
-          return true;
-        },
-        onDirExists: (dir) => {
-          // Don't overwrite existing sub-directories in node_modules
-          return false;
-        },
-      });
-    }
-  }
+  // make sure our directories are there
+  await fse.ensureDir(workingDir);
+  const buildDir = path.join(appDir, BUILD_DIR);
+  await fse.ensureDir(buildDir);
 
-  let output = {};
   if (!skipNpmInstall) {
-    if (printProgress) {
-      endSpinner();
-      startSpinner('Installing project dependencies');
-    }
-    output = await runCommand('npm', ['install', '--production'], {
-      cwd: tmpDir,
+    maybeStartSpinner('Copying project to temp directory');
+    const copyFilter = (src) => !src.endsWith('.zip');
+    await copyDir(appDir, workingDir, { filter: copyFilter });
+
+    maybeEndSpinner();
+    maybeStartSpinner('Installing project dependencies');
+    const output = await runCommand('npm', ['install', '--production'], {
+      cwd: workingDir,
     });
-  }
 
-  // `npm install` may fail silently without returning a non-zero exit code, need to check further here
-  const corePath = path.join(
-    tmpDir,
-    'node_modules',
-    constants.PLATFORM_PACKAGE,
-  );
-  if (!fs.existsSync(corePath)) {
-    throw new Error(
-      'Could not install dependencies properly. Error log:\n' + output.stderr,
-    );
+    // `npm install` may fail silently without returning a non-zero exit code,
+    // need to check further here
+    const corePath = path.join(workingDir, 'node_modules', PLATFORM_PACKAGE);
+    if (!fs.existsSync(corePath)) {
```

Comment:
```suggestion
    if (! await fse.pathExists(corePath)) {
```
For non-blocking IO performance